x-pack/plugins/ml/public/embeddables/anomaly_swimlane/initialize_swim_lane_data_fetcher.ts

```diff
@@ -96,6 +95,14 @@ export const initializeSwimLaneDataFetcher = (
 
         const { viewBy, swimlaneType, perPage, fromPage } = input;
 
+        let isEmptyQuery = true;
+
+        if (isOfQueryType(query)) {
+          isEmptyQuery = !query.query;
+        } else {
+          isEmptyQuery = !query?.esql;
+        }
+
```

Comment:
what if the query provides enough results for several pages, and the user wants to page further?

Reply:
I reckon we should approach it differently. When query _changes_ , we need to update `swimLaneInput$` observable, resetting the pagination.

Reply:
Changed in: [868dd88](https://github.com/elastic/kibana/pull/195090/commits/868dd88134912db5db1a07f42508217e6cbf7ca5)